src/gtclang/Frontend/CMakeLists.txt

```diff
@@ -43,3 +43,8 @@ mchbuild_add_library(
           StencilParser.h
   OBJECT
 )
+
+target_include_directories(GTClangFrontendObjects PRIVATE
+  $<TARGET_PROPERTY:Dawn::DawnShared,INTERFACE_INCLUDE_DIRECTORIES>
+)
```

Comment:
why is this one needed (just for interest)?

Reply:
This is because when you do something like
```#include "dawn.h" ``` 

the headers of dawn will be propagated to gtclang. 
For example if there is an include to Boost, then gtclang will also need to add those include directories. Even if gtclang does not know about Boost. 
So the way to solve it is by adding a set of include directories in the target of DawnShared which are going to be exported (i.e. using INTERFACE keyword). 
Only the directories that need to be exported to the dependent packages should be added. 
And then there is not need to find boost or specify boost dependency in gtclang. 
That is the modern cmake way of communicating dependencies via targets.